apps/studio/components/interfaces/Database/Triggers/TriggerSheet.tsx

```diff
@@ -127,6 +135,10 @@ export const TriggerSheet = ({ selectedTrigger, open, setOpen }: TriggerSheetPro
   })
   const { function_name, function_schema } = form.watch()
 
+  function isClosingSidePanel() {
+    form.formState.isDirty ? setIsClosingPanel(true) : onClose()
```

Comment:
It's a small one, but changing the select table doesn't seem to mark the form as dirty, meaning the user can still click out of the panel after changing the table (but nothing else).